src/client/metadataApiRetrieve.ts

```diff
@@ -30,15 +30,29 @@ export type MetadataApiRetrieveOptions = MetadataTransferOptions &
 
 export class RetrieveResult implements MetadataTransferResult {
   public readonly response: MetadataApiRetrieveStatus;
+  // The ComponentSet of retrieved components.
   public readonly components: ComponentSet;
 
-  constructor(response: MetadataApiRetrieveStatus, components: ComponentSet) {
+  // The ComponentSet of just the components on the local file system.
+  private localComponents: ComponentSet;
+  private fileResponses: FileResponse[];
+
+  constructor(
+    response: MetadataApiRetrieveStatus,
```

Comment:
can it use the constructor shorthand form?
https://www.typescriptlang.org/docs/handbook/typescript-tooling-in-5-minutes.html#classes